### DISCUSSION TITLE
Where to put auth?

### DISCUSSION THREAD

**Discussion by MincePie** [vercel/turborepo#4043] [Help] [answered]
Hi, I'm trying to get started with setting up a turborepo, but am a bit confused about how to configure the app (prisma, mongodb, react, tailwinds css and nextjs).

I used the template project (with tailwind) and have added a packages/database folder to add prisma, but am now confused about where to put next-auth. 

If I add these packages (https://next-auth.js.org/adapters/prisma) to the apps/web package.json, then where can I put the code that next-auth suggests I add to a lib/prismadb.ts and pages/api/auth/[...nextauth].js file?

Currently, my file structure is as follows: 

Thank you.
<img width="286" alt="Screen Shot 2023-03-02 at 12 02 15 pm" src="https://user-images.githubusercontent.com/10092303/222304002-c13fcaeb-abb4-4294-9d03-4c8b681929e8.png">

**Comment by anthonyshew** [comment, marked as answer]
You can package up `next-auth` exports that you need into an `auth` package for the rest of your repo to use. It will have a dependency on `database` so you can work with your database adapter inside the `auth` package.

You'll then export whatever utilities you need for your applications from your `auth` package. Voila, auth!
